tests/executors/test_kubernetes_executor.py

```diff
@@ -270,12 +270,27 @@ def test_change_state_skip_pod_deletion(self, mock_delete_pod, mock_get_kube_cli
                                             mock_kubernetes_job_watcher):
         executor = KubernetesExecutor()
         executor.kube_config.delete_worker_pods = False
+        executor.kube_config.delete_worker_pods_on_success = False
+
         executor.start()
         key = ('dag_id', 'task_id', 'ex_time', 'try_number2')
         executor._change_state(key, State.SUCCESS, 'pod_id')
         self.assertTrue(executor.event_buffer[key] == State.SUCCESS)
         mock_delete_pod.assert_not_called()
 
+    @mock.patch('airflow.executors.kubernetes_executor.KubernetesJobWatcher')
+    @mock.patch('airflow.executors.kubernetes_executor.get_kube_client')
+    @mock.patch('airflow.executors.kubernetes_executor.AirflowKubernetesScheduler.delete_pod')
+    def test_change_state_failed_pod_deletion(self, mock_delete_pod, mock_get_kube_client,
+                                            mock_kubernetes_job_watcher):
```

Comment:
This is causing static check failure: tests/executors/test_kubernetes_executor.py:285:45: E128 continuation line under-indented for visual indent